devito/mpi/halo_scheme.py

```diff
@@ -28,7 +28,39 @@ class HaloLabel(Tag):
 STENCIL = HaloLabel('stencil')
 
 
-HaloSchemeEntry = namedtuple('HaloSchemeEntry', 'loc_indices loc_dirs halos dims')
+class HaloSchemeEntry:
+
+    def __init__(self, loc_indices, loc_dirs, halos, dims):
```

Comment:
can it not inherit, alternatively, from EnrichedTuple?

Reply:
> that look redundant

with EnrichedTuple, once you properly override `__rargs__` and `__rkwargs__`, you probably don't need to override `__init__` and `__eq__`

Reply:
I fail to do so, so far,, the already existing code of DimensionTuple does not seem to help me....

Reply:
so what's the issue? 

There's a lot of redundant code here -- we should use EnrichedTuple

Then:

* you shouldn't need to override `__eq__`
* you shouldn't need to override `__repr__`
* I have some doubts about overriding `__hash__`, but maybe also see comment below

Reply:
I still do not understand this. SHould I still be able to construct HaloSchemeEntry the same way I do it now?
Will I need to specify getters ?
Like e.g. here:
```
    def glb_shape(self):
        """Shape of the decomposed domain."""
        return EnrichedTuple(*self._glb_shape, getters=self.dimensions)
```

Reply:
> SHould I still be able to construct HaloSchemeEntry the same way I do it now?

what happens if you try (so yes, w/o specifying any getters)